automated-testing/performance-testing/readme.md

```diff
@@ -0,0 +1,96 @@
+  
+## Sample performance testing scenario
+
+To understand performance testing better, let us consider a sample scenario where we needed to asses whether a web application could handle the expected production peak load in the coming months.
+The logical component view of the production application was as follows :
+![Production component view](./images/production-view.png)
+
+It was a simple micro services based web application. Some of the internal services consumed external third party services for their functionality.
+
+A Load test was planned for this application.
+
+As a part of the performance testing activities:
+
+* Success critieria for the test was defined
+* As a part of the test design activity a workload model for the application was created.
+* Load testing tool was finalized
+* Application infrastruture and application monitoring was enhanced to capture the required metrics
+* To minimise infrastructure cost associated with the performance testing it was decided to dynamically create and tear down the performance test environment
+![Performance Test Environment](./images/perf-test-view.png)
+* The performance test environment was configured as follows :
+  * The application which we need to test was deployed
+  * One of the external dependecies for this application provided a test endpoint, and the application in this environment was configured to hit that end point. The other external dependency did not provide a test endpoint so as part of the performance test activity scope an intellegent test stub was created which returns data in the expected format, response latency delays could be configured for this test stub.
+  * The load simulating components were deployed. These component/s simulated end user requests as per the workload profile which hit the Application UI component.
+* The initial load testing found resulted in the response time thresholds not being met. Analyis showed that issue was with one of the services. The issue was fixed and after re testing all success criteria were met.
+
+## Performance testing Frameworks and Tools
+
+Some of the performance testing tools are :
+
+* [JMeter](https://jmeter.apache.org/)
+* [Locust](https://locust.io/)
+* [Gatling](https://gatling.io/)
+  
+  The post "[Comparison of performance testing tools](https://www.blazemeter.com/blog/open-source-load-testing-tools-which-one-should-you-use?utm_source=blog&utm_medium=BM_blog&utm_campaign=jmeter-vs-locust-which-one-should-you-choose)" gives a nice comparison of some of the performance testing tools
+
+## Resources
+
+* Most of the guidance on this page is based on the awesome performance testing book [Patters and Practices: Performance Testing Guidance for Web Applications](https://docs.microsoft.com/en-us/archive/blogs/dajung/ebook-pnp-performance-testing-guidance-for-web-applications)
```

Comment:
Suggest not using awesome to descibe book, but more indicate its a good resource. Perhaps we could give more info on why it's a good resource.

Reply:
Incorporated the comment